x/mongo/driver/topology/server.go

```diff
@@ -446,12 +451,13 @@ func (s *Server) updateDescription(desc description.Server, initial bool) {
 		//  ¯\_(ツ)_/¯
 		_ = recover()
 	}()
-	s.desc.Store(desc)
 
-	callback, ok := s.updateTopologyCallback.Load().(func(description.Server))
+	// Use the updateTopologyCallback to update the parent Topology and get the description that should be stored.
+	callback, ok := s.updateTopologyCallback.Load().(updateTopologyCallback)
 	if ok && callback != nil {
```

Comment:
What does `ok == false` indicate to the caller?  Is `ok == false && callback != nil` possible?

Reply:
1. `ok == false` would either indicate nothing has been stored here, or that the stored thing is not of type `updateTopologyCallback`. Having nothing stored could be possible if only `NewServer` were called because `Connect` is what actually takes the callback, but `Connect` is also what starts the SDAM goroutine, so I don't think we could even get here if that were the case.

2. Given that, no I don't think `ok == false && callback != nil` is possible. It is possible that the callback is nil, as we do this in `Disconnect` to indicate that in-progress heartbeats should not update the topology.

You're probably wondering why the `ok` check is needed at all, and I'm not 100% sure, but I also don't want to change that in this PR.

Reply:
yeah, information request only.  Not asking for a change.  it seems harmless given the current state of the code.